x-pack/plugins/alerting/public/pages/maintenance_windows/components/table_actions_popover.tsx

```diff
@@ -0,0 +1,209 @@
+            }}
+            cancelButtonText={modalProps.cancelButton}
+            confirmButtonText={modalProps.title}
+            defaultFocusedButton="confirm"
+            buttonColor="danger"
+          >
+            <p>{modalProps.subtitle}</p>
+          </EuiConfirmModal>
+        );
+      }
+      return m;
+    }, [modalType, isModalVisible, closeModal, onArchive, onCancel, onCancelAndArchive]);
+
+    const items = useMemo(() => {
+      const menuItems = [];
+      if (status !== MaintenanceWindowStatus.Archived) {
+        menuItems.push(
+          <EuiContextMenuItem
+            data-test-subj="table-actions-edit"
+            key="edit"
+            icon="pencil"
+            onClick={() => {
+              closePopover();
+              onEdit();
+            }}
+          >
+            {i18n.TABLE_ACTION_EDIT}
+          </EuiContextMenuItem>
+        );
+      }
+      if (status === MaintenanceWindowStatus.Running) {
+        menuItems.push(
+          <EuiContextMenuItem
+            data-test-subj="table-actions-cancel"
+            key="cancel"
+            icon="stopSlash"
+            onClick={() => {
+              closePopover();
+              showModal('cancel');
+            }}
+          >
+            {i18n.TABLE_ACTION_CANCEL}
+          </EuiContextMenuItem>
+        );
+        menuItems.push(
+          <EuiContextMenuItem
+            data-test-subj="table-actions-cancel-and-archive"
+            key="cancel-and-archive"
+            icon="trash"
+            onClick={() => {
+              closePopover();
+              showModal('cancelAndArchive');
+            }}
+          >
+            {i18n.TABLE_ACTION_CANCEL_AND_ARCHIVE}
+          </EuiContextMenuItem>
+        );
+      }
+      if (
+        status !== MaintenanceWindowStatus.Running &&
+        status !== MaintenanceWindowStatus.Archived
+      ) {
+        menuItems.push(
+          <EuiContextMenuItem
+            data-test-subj="table-actions-archive"
+            key="archive"
+            icon="trash"
+            onClick={() => {
+              closePopover();
+              showModal('archive');
+            }}
+          >
+            {i18n.ARCHIVE}
+          </EuiContextMenuItem>
+        );
+      }
+      if (status === MaintenanceWindowStatus.Archived) {
+        menuItems.push(
+          <EuiContextMenuItem
+            data-test-subj="table-actions-unarchive"
+            key="unarchive"
+            icon="exit"
+            onClick={() => {
+              closePopover();
+              showModal('unarchive');
+            }}
+          >
+            {i18n.TABLE_ACTION_UNARCHIVE}
+          </EuiContextMenuItem>
+        );
+      }
+      return menuItems;
+    }, [status, onEdit, closePopover, showModal]);
+
+    return (
+      <>
+        <EuiFlexGroup justifyContent="flexEnd" alignItems="center">
+          <EuiFlexItem grow={false}>
+            <EuiPopover
+              button={
+                <EuiButtonIcon
```

Comment:
nitpick: useMemo or move to its own component

Reply:
Resolved in this commit https://github.com/elastic/kibana/pull/155201/commits/9d6e6730e18bd54c5bece8e97a0c801d232edd3c